### ISSUE TITLE
Add a `missing_error_doc` to `clippy::style`

### ISSUE THREAD

**Issue by lebensterben** [rust-lang/rust-clippy#6063]
### What it does

Checks for the doc comments of publicly visible functions that returns a `Result<T, E>` type and warns if there is no `# Errors` section.
Unsafe functions should document their safety preconditions, so that users can be sure they are using them safely.

### Categories (optional)

- Kind: `clippy::style`

### Why its this bad

Functions that returns a `Result<T, E>` type should document their on what conditions an `Err(E)` is returned, with the following benefits:

Makes code review easier. Though one can manually look up the documentation of
the errors the function may return, in practice this is very tedious because a
function may returns multiple different types that implement 
`std::error::Error` or equivalent traits from 3rd-party crates. A brief or 
detailed overview on the errors thus vastly increase the readability

#### Notes

`# Errors` section is already required in [Rust API Guidelines], though there
is no need to make this mandatory on all 3rd-party crates, the code-style of
standard library should be a good reference style.

### Drawbacks

None.

### Example

```rust
/// The error should really be documented
pub fn non_panic_u32_division(x: u32, y: u32) -> Result<u32, String> {
	if y == 0 {
		return Err(String::from("Divisor cannot be zero"))
	}
	x / y
}
```

Could be written as:

```rust
/// Computes `x / y`.
///
/// # Errors
///
/// Returns an error when the divisor is zero.
pub fn non_panic_u32_division(x: u32, y: u32) -> Result<u32, String> {
	if y == 0 {
		return Err(String::from("Divisor cannot be zero"))
	}
	x / y
}
```
P.S. This example looks simple enough. If you have any real-world example where the returned error is more complicated, please provide that in the comments.

[Rust API Guidelines]: https://rust-lang.github.io/api-guidelines/documentation.html#function-docs-include-error-panic-and-safety-considerations-c-failure

**Comment by flip1995** [comment]
This lint already exists, see [`missing_errors_doc`](https://rust-lang.github.io/rust-clippy/master/index.html#missing_errors_doc)
